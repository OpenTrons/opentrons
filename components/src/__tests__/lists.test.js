// list and list item components tests
import React from 'react'
import {MemoryRouter} from 'react-router'
import Renderer from 'react-test-renderer'

import {TitledList, ListItem, ListAlert, FLASK, CHECKED} from '..'

describe('TitledList', () => {
  test('adds an h3 with the title', () => {
    const heading = Renderer.create(
      <TitledList title='hello' />
    ).root.findByType('h3')

    expect(heading).toBeDefined()
    expect(heading.children).toEqual(['hello'])
  })

  test('adds an optional svg icon to title', () => {
    const icon = Renderer.create(
      <TitledList title='hello' iconName={FLASK} />
    ).root.findByType('svg')

    expect(icon).toBeDefined()
  })

  test('renders TitledList without icon correctly', () => {
    const tree = Renderer.create(
      <TitledList title='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('renders TitledList with children correctly', () => {
    const tree = Renderer.create(
      <TitledList title='foo'><li>Woop</li></TitledList>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('renders TitledList with optional icon correctly', () => {
    const tree = Renderer.create(
      <TitledList title='foo' icon={FLASK} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('renders expanded TitledList correctly', () => {
    const tree = Renderer.create(
      <TitledList onCollapseToggle={e => {}} description={<span>Description</span>} >
        <li>1</li>
        <li>2</li>
      </TitledList>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('renders collapsed TitledList correctly', () => {
    const tree = Renderer.create(
      <TitledList onCollapseToggle={e => {}} description={<span>Description</span>} collapsed>
        <li>1</li>
        <li>2</li>
      </TitledList>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})

describe('ListItem', () => {
  test('creates a linked list item from props', () => {
    const linkItemProps = {
      url: '/foo/bar',
      isDisabled: false
    }

    const root = Renderer.create(
      <MemoryRouter>
        <ListItem {...linkItemProps}>
          foo
        </ListItem>
      </MemoryRouter>
    ).root

    const link = root.findByType('a')
    expect(link.props.href).toBe(linkItemProps.url)
    expect(link.props.disabled).toBe(false)
  })

  test('adds an optional svg icon as child', () => {
    const icon = Renderer.create(
      <ListItem iconName={CHECKED} />
    ).root.findByType('svg')

    expect(icon).toBeDefined()
  })

  test('renders ListItem with icon correctly', () => {
    const tree = Renderer.create(
      <ListItem to='/hello' iconName={CHECKED} isDisabled='false' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('renders ListItem without icon correctly', () => {
    const tree = Renderer.create(
      <ListItem to='/hello' isDisabled='false' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})

describe('ListAlert', () => {
  test('list alert renders correctly', () => {
    const tree = Renderer.create(
      <ListAlert> alert alert</ListAlert>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
