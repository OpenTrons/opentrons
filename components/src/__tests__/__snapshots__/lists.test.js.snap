// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListAlert list alert renders correctly 1`] = `
<li
  className="alert"
>
   alert alert
</li>
`;

exports[`ListItem renders ListItem with icon correctly 1`] = `
<li
  onClick={undefined}
>
  <svg
    aria-hidden="true"
    className="icon"
    fill="currentColor"
    height={undefined}
    version="1.1"
    viewBox="0 0 24 24"
    width={undefined}
    x={undefined}
    y={undefined}
  >
    <path
      d="M9.6,18 L3.6,12 L5.292,10.296 L9.6,14.604 L18.708,5.496 L20.4,7.2 L9.6,18 Z M12,5.32907052e-16 C5.372583,-2.13162821e-15 5.32907052e-16,5.372583 0,12 C-1.59872116e-15,15.1825979 1.26428208,18.2348448 3.51471863,20.4852814 C5.76515517,22.7357179 8.81740212,24 12,24 C15.1825979,24 18.2348448,22.7357179 20.4852814,20.4852814 C22.7357179,18.2348448 24,15.1825979 24,12 C24,8.81740212 22.7357179,5.76515517 20.4852814,3.51471863 C18.2348448,1.26428208 15.1825979,2.66453526e-16 12,5.32907052e-16 Z"
      fillRule="evenodd"
    />
  </svg>
  <span
    className="info"
  />
</li>
`;

exports[`ListItem renders ListItem without icon correctly 1`] = `
<li
  onClick={undefined}
>
  <span
    className="info"
  />
</li>
`;

exports[`TitledList renders TitledList with children correctly 1`] = `
<div
  className=""
>
  <div
    className="list_title_bar"
    onClick={undefined}
  >
    <h3
      className="list_title"
    >
      foo
    </h3>
  </div>
  <ol
    className="titled_list"
  >
    <li>
      Woop
    </li>
  </ol>
</div>
`;

exports[`TitledList renders TitledList with optional icon correctly 1`] = `
<div
  className=""
>
  <div
    className="list_title_bar"
    onClick={undefined}
  >
    <h3
      className="list_title"
    >
      foo
    </h3>
  </div>
</div>
`;

exports[`TitledList renders TitledList without icon correctly 1`] = `
<div
  className=""
>
  <div
    className="list_title_bar"
    onClick={undefined}
  >
    <h3
      className="list_title"
    >
      foo
    </h3>
  </div>
</div>
`;

exports[`TitledList renders collapsed TitledList correctly 1`] = `
<div
  className=""
>
  <div
    className="list_title_bar"
    onClick={undefined}
  >
    <h3
      className="list_title"
    />
    <div
      className="accordion_carat"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className=""
        fill="currentColor"
        height={undefined}
        version="1.1"
        viewBox="0 0 24 24"
        width={undefined}
        x={undefined}
        y={undefined}
      >
        <path
          d="M7.41 9L12 13.242 16.59 9 18 10.312l-6 5.545-6-5.545z"
          fillRule="evenodd"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`TitledList renders expanded TitledList correctly 1`] = `
<div
  className=""
>
  <div
    className="list_title_bar"
    onClick={undefined}
  >
    <h3
      className="list_title"
    />
    <div
      className="accordion_carat"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className=""
        fill="currentColor"
        height={undefined}
        version="1.1"
        viewBox="0 0 24 24"
        width={undefined}
        x={undefined}
        y={undefined}
      >
        <path
          d="M8.571 7.839l4.242 4.59-4.242 4.59 1.312 1.41 5.545-6-5.545-6z"
          fillRule="evenodd"
        />
      </svg>
    </div>
  </div>
  <span>
    Description
  </span>
  <ol
    className="titled_list"
  >
    <li>
      1
    </li>
    <li>
      2
    </li>
  </ol>
</div>
`;
